### DISCUSSION TITLE
Comparison of speed and VRAM of different attention optimizations

### DISCUSSION THREAD

**Discussion by sgsdxzy** [d8ahazard/sd_dreambooth_extension#230] [Ideas]
### Settings:
Card: 3080Ti 12288MiB VRAM
![image](https://user-images.githubusercontent.com/1655353/202113801-ac22fb41-f039-4b54-88d4-fff420efbbf3.png)
cached latents of 7 samples, batch size 1

### Results:
| Attention  | Speed(it/s) | VRAM Usage(MiB) by nvidia-smi |
| ------------- | ------------- | ------------- |
| Default  | 3.57  | 11684 |
| xformers | 2.97  | 10424 |
| FlashAttention | 2.37  | 10274 |

As far as I know xformers integrated FlashAttention. 
I would recommand xformers for 12GB because you can use larger batch size.

**Comment by 0xItx** [comment]
Any idea why xformers & FlashAttention cause such a speed degradation?
